python/paddle/v2/dataset/common.py

```diff
@@ -74,3 +76,59 @@ def fetch_all():
             getattr(
                 importlib.import_module("paddle.v2.dataset.%s" % module_name),
                 "fetch")()
+
+def split(reader, line_count, suffix="%05d.pickle"):
+    """
+    you can call the function as:
+
+    split(paddle.v2.dataset.cifar.train10(), line_count=1000,
+        suffix="imikolov-train-%05d.pickle")
+
+    the output files as:
+
+    |-imikolov-train-00000.pickle
+    |-imikolov-train-00001.pickle
+    |- ...
+    |-imikolov-train-00480.pickle
+
+    :param reader: the reader creator will be split
+    :param line_count: line count for each file
+    :param suffix: the suffix for each file,
+                    contain "%d" means the id for each file
+    """
+    lines = []
+    indx_f = 0
+    for i, d in enumerate(reader()):
+        lines.append(d)
+        if i >= line_count and i % line_count == 0:
+            with open(suffix % indx_f, "w") as f:
+                pickle.dump(lines, f)
+                lines = []
+                indx_f += 1
+    if not lines:
+        with open(suffix % indx_f, "w") as f:
+            pickle.dump(lines, f)
```

Comment:
Maybe we should add a more generic `dump` function, or accept a dump function as callback, so that we can split the files to more types like `RecordIO` or `protobuf`